Library/Formula/gpg-agent.rb

```diff
@@ -33,9 +33,46 @@ def install
     system "make", "install"
   end
 
+  def caveats; <<-EOS.undent
+      To replace ssh-agent with gpg-agent, you need to:
+         * ensure that "enable-ssh-support" is in ~/.gnupg/gpg-agent.conf
+         * ensure that "pinentry" in ~/.gnupg/gpg-agent.conf has a graphical PIN entry
+           such as pinentry-mac (needed for GUI programs)
+         * disable the system ssh-agent with
+             launchctl unload -w /System/Library/LaunchAgents/org.openssh.ssh-agent.plist
+         * have launchd start gpg-agent at login
+    EOS
+  end
+
   test do
     system "#{bin}/gpg-agent", "--help"
   end
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+            <string>/bin/sh</string>
+            <string>-c</string>
+            <string>#{opt_prefix}/bin/gpg-agent -c --daemon | /bin/launchctl</string>
```

Comment:
gpg-agent -c --daemon outputs several csh-format setenv lines, to set the needed environmental variables to allow other programs to use the gpg-agent and it's emulated ssh-agent functionality.  These are GPG_AGENT_INFO, SSH_AUTH_SOCK, and SSH_AGENT_PID.  By piping these commands to launchctl, they get executed and stuck into the user's launchd environment.  Thus they become available to any terminal windows open, but ALSO to any gui apps that are launched, so things like CyberDuck, or SourceTree, or anything else that needs SSH support can use the gpg-agent.

```
# launch env not set
lgml-ryand:homebrew ryand$ launchctl export | egrep 'GPG|SSH'
# start it normally to look at the output
lgml-ryand:homebrew ryand$ gpg-agent -c --daemon
setenv GPG_AGENT_INFO /Users/ryand/.gnupg/S.gpg-agent:2115:1;
setenv SSH_AUTH_SOCK /Users/ryand/.gnupg/S.gpg-agent.ssh;
setenv SSH_AGENT_PID 2115;

# launch env still not set
lgml-ryand:homebrew ryand$ launchctl export | egrep 'GPG|SSH'

#restart gpg-agent, put the agent vars into the launch environment
lgml-ryand:homebrew ryand$ pkill gpg-agent
lgml-ryand:homebrew ryand$ gpg-agent -c --daemon | launchctl

# and now they are set
lgml-ryand:homebrew ryand$ launchctl export | egrep 'GPG|SSH'
GPG_AGENT_INFO="/Users/ryand/.gnupg/S.gpg-agent:2266:1;"; export GPG_AGENT_INFO;
SSH_AUTH_SOCK="/Users/ryand/.gnupg/S.gpg-agent.ssh;"; export SSH_AUTH_SOCK;
SSH_AGENT_PID="2266;"; export SSH_AGENT_PID;
```

The variables can change every start; gpg-agent can use a randomized path in addition to the standard path that I'm using, plus the GPG_AGENT_INFO contains the running pid for the agent.

Putting them into the launchd environment is the only way to let the GUI apps use gpg-agent.

Longer term, I'm working on a patch to submit upstream to add launchd support directly into gpg-agent (which they may still reject, as it's very osx-specific), but this is the short-term fix for those of us who use gpg-agent (in place of ssh-agent, or alongside).